### PR TITLE
Generate feedback docs on config errors

### PR DESCRIPTION
**Pull request by garethbowen**
Catch errors in configuration early and submit them as Feedback
docs so app builders can find and fix them early.

medic/cht-core#6629

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: shared-libs/rules-engine/src/rules-emitter.js [hunk 1]
```diff
@@ -145,6 +152,9 @@ const startSession = function() {
       session.match(err => {
         session.dispose();
         if (err) {
+          if (feedbackService) {
```
Comment (craig-landry) [inline shared-libs/rules-engine/src/rules-emitter.js:155]: This is probably a better place to make my null object request.... the conditional nesting here might easily be incorrectly "optimized" to be on a single line with an `&&` but of course that would create a bug. Removing this conditional steers clear of that entirely.

File: shared-libs/rules-engine/src/rules-emitter.js [hunk 2]
```diff
@@ -51,6 +55,9 @@ module.exports = {
         },
       });
     } catch (err) {
+      if (feedbackService) {
```
Comment (craig-landry) [inline shared-libs/rules-engine/src/rules-emitter.js:58]: What's the case where this is missing? I'd prefer that we can ensure it is always present and remove the conditional. Even if it is a legitimate case that it not be injected, my preference is still to ensure its presence via the null object pattern where up in the constructor we'd set the null object as a default like this:

```
  initialize: (settings, Feedback = new NullFeedbackService()) => {
```

**Review by craig-landry — COMMENTED** [review]
I don't see anything objectionable here, aside from my comments on my null object preference. I'm happy to talk though options and preferences on that.

**Comment by garethbowen** [comment]
@craig-landry Thanks for the review! I've gone with the NullFeedbackService pattern. Initially this lib was designed so it could be used by integration tests for configurations so they wouldn't have to launch the whole app. I'm not sure if it ever happened but I wanted to make it backwards compatible just in case...

**Review by craig-landry — APPROVED** [review]

**Comment by garethbowen** [comment]
This code has been ported and merged into the angular 10 migration branch. Closing this PR.

**Pull request closed without merging**